packages/server-wallet/src/protocols/channel-closer.ts

```diff
@@ -31,40 +31,33 @@ export class ChannelCloser {
   public async crank(objective: DBCloseChannelObjective, response: WalletResponse): Promise<void> {
     const channelToLock = objective.data.targetChannelId;
 
-    await this.store.lockApp(channelToLock, async (tx, channel) => {
-      if (!channel) {
-        throw new Error('Channel must exist');
-      }
-
-      try {
+    await this.store
+      .lockApp(channelToLock, async (tx, channel) => {
         if (!ensureAllAllocationItemsAreExternalDestinations(channel)) {
           response.queueChannel(channel);
           return;
         }
 
+        if (!(await this.areAllFinalStatesSigned(channel, tx, response))) {
+          response.queueChannel(channel);
+          return;
+        }
+
         const defunder = Defunder.create(
           this.store,
           this.chainService,
           this.logger,
           this.timingMetrics
         );
 
-        if (!(await this.areAllFinalStatesSigned(channel, tx, response))) {
-          response.queueChannel(channel);
-          return;
-        }
-
         if (!(await defunder.crank(channel, tx)).isChannelDefunded) {
           response.queueChannel(channel);
           return;
         }
 
         await this.completeObjective(objective, channel, tx, response);
-      } catch (error) {
-        this.logger.error({error}, 'Error taking a protocol step');
-        await tx.rollback(error);
-      }
-    });
+      })
+      .catch(err => this.logger.error({err, objective}, 'Cannot crank ChannelCloser'));
```

Comment:
The changes to this file seem like a good refactor, since we don't need to call `tx.rollback()` manually when an error is thrown (it happens automatically). 

However, the changes to the other files are introducing new behaviour which is potentially quite important: errors are being caught and not thrown. Is that an intended change in behaviour? If so what is the rationale?

If it is intended, I would be in favour of an approach [like this one ](https://github.com/statechannels/statechannels/pull/3185/files?diff=split&w=1). In other words, I would :
 -  reduce replication by putting the error handling one layer above (in the objective manager)
 - highlight the fact that errors are not re-thrown, and explain if possible, in a comment

Reply:
Yep, this change wasn't intended to change behaviour, though it obviously does.

It was intended to be a quick and easy-to-approve change. Since it isn't, I'm going to close this PR.